apps/cms/scripts/import-data/import.mjs

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+
+import 'dotenv/config'
+
+import FormData from 'form-data'
+import { createReadStream, writeFileSync } from 'fs'
+import { resolve as _resolve } from 'path'
+
+import {
+  checkCsvAsync,
+  createXlsFile,
+  getCMSAuthToken,
+  getConfigFromStdin,
+  groupFilesFromDirectoryByExtension,
+  importDataFile,
+  parseCsvData,
+  readFileAsync,
+  removeFile,
+  updateCsvAsync,
+} from '../utils.mjs'
+
+async function main(args) {
+  /**
+   * Read config file if it exists, or ask user to provide it.
+   * Once credentials are provided, get an auth token from directus.
+   */
+  console.log('Authenticating...')
+  let config = { email: '', password: '' }
+  if (args.length < 3) {
+    config = await getConfigFromStdin()
+  } else {
+    config = JSON.parse(await readFileAsync(_resolve(args[2])))
+  }
+  const token = await getCMSAuthToken(config)
+
+  /**
+   * Begin importing data.
+   */
+  console.log('Starting import...')
+  const basePath = './scripts/import-data'
+  const files = await groupFilesFromDirectoryByExtension(basePath, 'csv')
+  // Loop through the import data
+  for (const file of files) {
+    const csvFile = _resolve(`${basePath}/${file}`)
```

Comment:
super minor, but if the order doesn't matter you could probably

```javascript
const processFile = async (file, basePath, token) => {
  const csvFile = _resolve(`${basePath}/${file}`)
  const collection = file.split('.').shift()
  console.log(`Checking file data for ${collection}...`)
  // Parse CSV and check values.
  const csvData = await parseCsvData(csvFile)
  await checkCsvAsync(csvData, collection, token)
  // Create images and update file.
  const data = await updateCsvAsync(csvData, basePath, token)
  const formData = new FormData()
  const jsonFile = `${basePath}/${collection}.json`
  // Create JSON file.
  writeFileSync(jsonFile, JSON.stringify(data))
  formData.append('file', createReadStream(jsonFile))
  // Import data to CMS
  console.log(`Importing file for ${collection}...`)
  await importDataFile(formData, collection, token)
  // Remove newly created JSON file
  await removeFile(jsonFile)
}
```

```javascript
const promises = []
  // Loop through the import data
  for (const file of files) {
    promises.push(processFile(file, basePath, token))
  }
  await Promise.all(promises)

  console.log('Done!')
```

to slightly speed it up

Reply:
Good idea! Currently there's no order except alphabetical with the CSV files so this would work great. Long-term it'd be great if we could import in order and reference new items being created in the import process, but there would be some complexity there. Going to test a bit but for the current setup this should be perfect.

Reply:
Ah actually found an issue. It tries to execute the readlineAsync function at the same time, so as you type image values it writes every letter multiple times.